### ISSUE TITLE
 Rule for detecting unneccessary braces aroung the trailing lambda

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6008]
## Expected Rule behaviour
The rule should flag that the trailing lambda can be used without the braces around it. Examples of compliant and non-compliant code

```
// non-compliant code
fun test() {
    repeat(10, {
       println(it)
    })
}

// compliant code
fun test() {
    repeat(10) {
        println(it)
    }
}
```

This rule as mentioned at https://github.com/pinterest/ktlint/issues/1949#issuecomment-1511917556 can't be implemented by `ktlint` as it required binding context.

**Comment by BraisGabin** [comment]
I like the idea, clearly a style rule.

Some configurations would be needed. For example, don't flag if the lambda is a named parameter. Or allow trailing lambda if the previous parameter is another lambda.

**Comment by luanpotter** [comment]
There is also this (possibly simpler) bad case:

```kotlin
fun test() {
    foo() {
        println(it)
    }
}
```

(ie unnecessary parenthesis when the lambda is already outside and there are no other arguments)

**Comment by atulgpt** [comment]
> 

This we already detect why a wrapped KtLint rule `UnnecessaryParenthesesBeforeTrailingLambda`
